rsts/index.rst

```diff
@@ -54,27 +54,208 @@ Created at `Lyft <https://www.lyft.com/>`__ in collaboration with Spotify, Freen
+              override def run(input: GreetTaskInput): GreetTaskOutput = GreetTaskOutput(s"Welcome, ${input.name}!")
+            }
+
+            object GreetTask {
+              /**
+               * Binds input data to this task
+               *
+               * @param name the input name
+               * @return 1 transformed instance of this class with input data
+               */
+              def apply(name: SdkBindingData): SdkTransform =
+                new GreetTask().withInput("name", name)
+            }
+
+        You can compose one or more tasks to create a ``workflow``:
+
+        .. code:: scala
+
+            case class AddQuestionTaskInput(greeting: String)
+            case class AddQuestionTaskOutput(greeting: String)
+
+            /**
+             * Example Flyte task that takes a greeting message as input, appends "How are you?", and outputs
+             * the result
+             */
+            class AddQuestionTask
+                extends SdkRunnableTask(
+                  SdkScalaType[AddQuestionTaskInput],
+                  SdkScalaType[AddQuestionTaskOutput]
+                ) {
+
+              /**
+               * Defines task behavior. This task takes a greeting message as the input, append it with " How
+               * are you?", and outputs a new greeting message.
+               *
+               * @param input the greeting message
+               * @return the updated greeting message
+               */
+              override def run(input: AddQuestionTaskInput): AddQuestionTaskOutput = AddQuestionTaskOutput(s"${input.greeting} How are you?")
+            }
+
+            object AddQuestionTask {
+              /**
+               * Binds input data to this task
+               *
+               * @param greeting the input greeting message
+               * @return a transformed instance of this class with input data
+               */
+              def apply(greeting: SdkBindingData): SdkTransform =
+                new AddQuestionTask().withInput("greeting", greeting)
+            }
+
+        .. code:: scala
+
+            class WelcomeWorkflow extends SdkWorkflow {
+
+              def expand(builder: SdkWorkflowBuilder): Unit = {
+                // defines the input of the workflow
+                val name = builder.inputOfString("name", "The name for the welcome message")
+
+                // uses the workflow input as the task input of the GreetTask
+                val greeting = builder.apply("greet", GreetTask(name)).getOutput("greeting")
+
+                // uses the output of the GreetTask as the task input of the AddQuestionTask
+                val greetingWithQuestion = builder.apply("add-question", AddQuestionTask(greeting)).getOutput("greeting")
+
+                // uses the task output of the AddQuestionTask as the output of the workflow
+                builder.output("greeting", greetingWithQuestion, "Welcome message")
+              }
+            }
```

Comment:
Is there something similar to calling it like in python? https://github.com/flyteorg/flyte/pull/1183/files#diff-9a1147ef4c0aeaf0680900dbd2902c51d2a5475c0c21a606fbe1c9420c4261edR80-R81

```
           welcome("Traveler")
           # Output: "Welcome, Traveler! How are you?"
```

Reply:
Yes, like a main() method

Reply:
Turns out that although it's easy to execute a task by itself, it's not trivial to execute a workflow similar to Python.